src/components/Badge/Badge.tsx

```diff
@@ -0,0 +1,41 @@
+import InteractiveLink from 'components/core/InteractiveLink/InteractiveLink';
+import Tooltip from 'components/Tooltip/Tooltip';
+import { useCallback, useState } from 'react';
+import styled from 'styled-components';
+
+type Props = {
+  name: string;
+  imageURL: string;
+};
+
+export default function Badge({ name, imageURL }: Props) {
+  const [showTooltip, setShowTooltip] = useState(false);
+  const handleMouseEnter = useCallback(() => {
+    setShowTooltip(true);
+  }, []);
+
+  const handleMouseExit = useCallback(() => {
+    setShowTooltip(false);
+  }, []);
+
+  // TODO: replace with badge contract address
+  const communityUrl = '/community/0x31982936afe7fc432ec1a2766bdba4d07a779b30';
```

Comment:
@Robinnnnn What's the best way to get contract address art gobblers for community page?

Seems like their collection used opensea storefront - https://opensea.io/collection/art-gobbler

Reply:
hmm good question, so i think the badge should come with the contract address for us to link to it. this means we'll need to update the backend schema to return the contract:

<img width="182" alt="image" src="https://user-images.githubusercontent.com/12162433/192689670-c82f5e9e-a9d2-4379-85a7-0fb8164e0d9a.png">

^ this just returns an ID, but we the schema should really be `contract: Contract`. will ask someone to implement this tomorrow.

Reply:
in the meantime, best practice graphql patterns is to pull the badge data as far down the react tree as possible – so `{ name, imageURL }: Props` should really be `{ badgeRef }: Props`

more expanded:
```ts
type Props = {
    // this type will be auto-generated by Relay once you hit "save".
    badgeRef: BadgeFragment$key 
}

export default function Badge({ badgeRef }: Props) {
    const badge = useFragment(
        graphql`
            fragment BadgeFragment on Badge {
                name
                imageURL
                contract {
                    address
                }
            }
        `,
        badgeRef
    )

    const { name, imageURL, contract: { address }} = badge;

    ...
}

// parent component
const user = useFragment(
    graphql`
      fragment UserGalleryHeaderFragment on GalleryUser {
        username @required(action: THROW)
        dbid
        bio
        badges {
          name
          ...BadgeFragment <-- main change
        }
      }
    `,
    userRef
  );

  ...

  {user.badges.map(badge => <Badge key={badge.name} badgeRef={badge} />)}

```

Reply:
TIL. Thanks a lot @Robinnnnn for the detail explanation!